src/ledger/NetworkConfig.h

```diff
@@ -0,0 +1,203 @@
+// under the Apache License, Version 2.0. See the COPYING file at the root
+// of this distribution or at http://www.apache.org/licenses/LICENSE-2.0
+
+#include <cstdint>
+#include <ledger/LedgerTxn.h>
+
+namespace stellar
+{
+
+// Defines the initial values of the network configuration
+// settings that are applied during the protocol version upgrade.
+// These values should never be changed after the protocol upgrade
+// happens - any further changes should be performed separately via
+// config upgrade mechanism.
+struct InitialNetworkConfig
+{
+    // Contract size settings
+    static constexpr uint32_t MAX_CONTRACT_SIZE = 65536;
+
+    // Compute settings
+    static constexpr int64_t LEDGER_MAX_INSTRUCTIONS = 1;
+    static constexpr int64_t TX_MAX_INSTRUCTIONS = 1;
+    static constexpr int64_t FEE_RATE_PER_INSTRUCTIONS_INCREMENT = 1;
+    static constexpr uint32_t MEMORY_LIMIT = 1;
+
+    // Ledger access settings
+    static constexpr uint32_t LEDGER_MAX_READ_LEDGER_ENTRIES = 1;
+    static constexpr uint32_t LEDGER_MAX_READ_BYTES = 1;
+    static constexpr uint32_t LEDGER_MAX_WRITE_LEDGER_ENTRIES = 1;
+    static constexpr uint32_t LEDGER_MAX_WRITE_BYTES = 1;
+    static constexpr uint32_t TX_MAX_READ_LEDGER_ENTRIES = 1;
+    static constexpr uint32_t TX_MAX_READ_BYTES = 1;
+    static constexpr uint32_t TX_MAX_WRITE_LEDGER_ENTRIES = 1;
+    static constexpr uint32_t TX_MAX_WRITE_BYTES = 1;
+    static constexpr int64_t FEE_READ_LEDGER_ENTRY = 1;
+    static constexpr int64_t FEE_WRITE_LEDGER_ENTRY = 1;
+    static constexpr int64_t FEE_READ_1KB = 1;
+    static constexpr int64_t FEE_WRITE_1KB = 1;
+    static constexpr int64_t BUCKET_LIST_SIZE_BYTES = 1;
+    static constexpr int64_t BUCKET_LIST_FEE_RATE_LOW = 1;
+    static constexpr int64_t BUCKET_LIST_FEE_RATE_HIGH = 1;
+    static constexpr uint32_t BUCKET_LIST_GROWTH_FACTOR = 1;
+
+    // Historical data settings
+    static constexpr int64_t FEE_HISTORICAL_1KB = 1;
+
+    // Bandwidth settings
+    static constexpr uint32_t LEDGER_MAX_PROPAGATE_SIZE_BYTES = 1;
+    static constexpr uint32_t TX_MAX_SIZE_BYTES = 1;
+    static constexpr int64_t FEE_PROPAGATE_DATA_1KB = 1;
+
+    // Meta data settings
+    static constexpr uint32_t TX_MAX_EXTENDED_META_DATA_SIZE_BYTES = 1;
+    static constexpr int64_t FEE_EXTENDED_META_DATA_1KB = 1;
+
+    // Host logic version
+    static constexpr uint32_t HOST_LOGIC_VERSION = 1;
+};
+
+// Wrapper for the contract-related network configuration.
+class ContractNetworkConfig
+{
+  public:
+    // Creates the initial contract configuration entries for protocol v20.
+    // This should happen once during the correspondent protocol version
+    // upgrade.
+    static void createLedgerEntriesForV20(AbstractLedgerTxn& ltx);
+    // Test-only function that initializes contract network configuration
+    // bypassing the normal upgrade process (i.e. when genesis ledger starts not
+    // at v1)
+    static void
```

Comment:
This should be wrapped in a `BUILD_TESTS` `ifdef` like https://github.com/stellar/stellar-core/blob/bff2c2b37402acc61166c0f369da1205efbe8c9d/src/bucket/BucketManagerImpl.h#L147-L155

Reply:
`USE_CONFIG_FOR_GENESIS` is not conditioned by this macro (probably it should be), so I can't use it here as well. I suppose we support this for standalone runs of core with bumped protocol version.

Reply:
`USE_CONFIG_FOR_GENESIS` is not conditioned behind any macros, so wouldn't it still work?

Reply:
I mean that we need to call this when `USE_CONFIG_FOR_GENESIS ` is set to true. I'm not entirely sure why is it not guarded by `BUILD_TESTS` but for now in order to make the code compile without `BUILD_TESTS` we can't use this condition for my function.